crates/ibc/src/core/ics24_host/identifier.rs

```diff
@@ -131,19 +117,38 @@ impl ChainId {
     /// assert_eq!(ChainId::new("chainA", 1).with_version(2), ChainId::new("chainA", 2));
     /// assert_eq!("chain1".parse::<ChainId>().unwrap().with_version(2), "chain1".parse::<ChainId>().unwrap());
     /// ```
-    pub fn with_version(mut self, version: u64) -> Self {
-        if Self::is_epoch_format(&self.id) {
-            self.id = {
-                let mut split: Vec<&str> = self.id.split('-').collect();
-                let version = version.to_string();
-                if let Some(last_elem) = split.last_mut() {
-                    *last_elem = &version;
-                }
-                split.join("-")
-            };
-            self.version = version;
+    pub fn with_version(self, version: u64) -> Self {
+        if self.version == 0 {
+            self
+        } else {
+            let (name, _) = Self::split_chain_id(&self.id).unwrap();
```

Comment:
Can here we leave with `expect` instead of `unwrap()`?

Reply:
Honestly I don’t see the point of `expect` method.  Presumably it gives more context about the failure except the method is never used that way.  In this repository, 72% uses of `expect` simply use `Never fails` as error message.  That’s just pointless noise and in those places `unwrap` would be better.

Instead I’ve added a comment explaining why the unwrap never fails.

Reply:
Oh Christ, I see you guys enabled this in Clippy. Fine then.

Reply:
Indeed. We definitely should do refinements around our `expect`, and even a bunch of error messages.